registry/storage/schema2manifesthandler.go

```diff
@@ -80,7 +83,26 @@ func (ms *schema2ManifestHandler) verifyManifest(ctx context.Context, mnfst sche
 		}
 
 		for _, fsLayer := range mnfst.References() {
-			_, err := ms.repository.Blobs(ctx).Stat(ctx, fsLayer.Digest)
+			var err error
+			if fsLayer.MediaType != schema2.MediaTypeForeignLayer {
+				if len(fsLayer.URLs) == 0 {
+					_, err = ms.repository.Blobs(ctx).Stat(ctx, fsLayer.Digest)
+				} else {
+					err = errors.New("unexpected URL on layer")
+				}
+			} else {
+				// Clients download this layer from an external URL, so do not check for
+				// its presense.
+				if len(fsLayer.URLs) == 0 {
+					err = errors.New("missing URL on layer")
+				}
+				for _, u := range fsLayer.URLs {
+					pu, err := url.ParseRequestURI(u)
+					if err != nil || (pu.Scheme != "http" && pu.Scheme != "https") {
+						err = errors.New("invalid URL on layer")
```

Comment:
Whoops, good catch. This is by far the most common mistake I make in Go...

I'll add a test.